### PR TITLE
[doc] Register migration to Cozy v3

### PR DESCRIPTION
**Pull request by gregorylegarec**

### CODE REVIEW

File: docs/client/migration-to-cozy-v3.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+instance = {
+  publicName: 'Alice Martin',
+  ownerEmail: 'alice@example.com',
+  tz: 'Europe/Paris'
+}
+```
+> See [here](https://github.com/cozy/cozy-proxy/blob/master/client/app/config/steps/infos.coffee#L25) and [here](https://github.com/cozy/cozy-proxy/blob/master/client/app/config/steps/infos.coffee#L37) for example.
+
+## Handle registerToken
+Presence of a `registerToken` means that the user access to her Cozy for the first time.
```
Comment (GoOz) [inline docs/client/migration-to-cozy-v3.md]: I guess you should say 'her/his' or 'her or his', don't you think?

File: docs/client/migration-to-cozy-v3.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                            @handleStepChanged(step)
+                        onStepFailed: (step, err) =>
+                            @handleStepFailed(step, err)
+                        onDone: () =>
+                            @handleTriggerDone()
+```
+The initial onboardedSteps are disappearing to be fetched later in the process.
+
+## Dealing without saving first steps
+With Cozy v3, we cannot save the step progression anymore, at least for the steps before the passphrase step. So both `welcome` and `agreement` steps should not have a `save` method anymore (So if the user leaves the onboarding before completing the `password` step, she will have to start from `welcome` step the next time she will access her Cozy).
```
Comment (GoOz) [inline docs/client/migration-to-cozy-v3.md]: Same as before. her/his and 'she/he'

File: docs/client/migration-to-cozy-v3.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+## Display the total number of steps
+In the current onboarding, we are using a property `hasValidInfos` to determine how many steps the onboarding will have. This property is computed and transmitted by the server. We cannot compute it client side without having access to `public-name` and `owner-email`, which should imply a serious security flaw.
+
+Not having this property transmitted by the server will force us to display an estimated number of steps, or display this number only after the `password` step. This point needs to be clarified with both Backend Team and Product Team.
+
+Maybe be the `hasValidInfos` property could be encrypted in the `registerToken` ?
+
+## Handle contextToken
+Presence of `contextToken` means that the user is already authentified. In this case, the onboarding should fetch instance data by performing a `GET` on the endpoint `/instance`. Once data is fetched, especially the `onboardedSteps` property, the onboarding determines the uncompleted steps or redirect the user to her home, as it currently does.
```
Comment (GoOz) [inline docs/client/migration-to-cozy-v3.md]: same. her/his

File: docs/client/migration-to-cozy-v3.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+## Handle missing tokens
+If none of `registerToken` or `contextToken` are passed to the onboarding, we have to throw an error and display a dedicated error page. This is some new stuff which has to be designed.
+This behavior allows us to ignore the case when the user access her Cozy without `registerToken`, as we are not able to detect if a passphrase exists on the Cozy instance.
```
Comment (GoOz) [inline docs/client/migration-to-cozy-v3.md]: same. her/his

**Comment by gregorylegarec** [comment]
@GoOz Actually I used the same user as the one in https://github.com/cozy/cozy-stack/blob/master/docs/onboarding.md which is Alice, but I may change that.

**Comment by GoOz** [comment]
@gregorylegarec Ah ok. Never occured to me, sorry! 😛

**Comment by CPatchane** [comment]
Thanks @gregorylegarec!

**Pull request merged**